### PR TITLE
Cargo - Add setting for carry after cargo unload

### PR DESCRIPTION
**Pull request by tbeswick96**
**When merged this pull request will:**
- Add a setting to disable automatically carrying/dragging cargo on unload

### IMPORTANT

- [x] If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [x] [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- [x] Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/cargo/initSettings.sqf [hunk 1]
```diff
@@ -42,3 +42,12 @@
     false,
     {[QGVAR(enableRename), _this, true] call EFUNC(common,cbaSettings_settingChanged)}
 ] call CBA_fnc_addSetting;
+
+[
+    QGVAR(carryAfterUnload), "CHECKBOX",
+    [LSTRING(carryAfterUnload), LSTRING(carryAfterUnload_description)],
+    [LELSTRING(OptionsMenu,CategoryLogistics), LLSTRING(openMenu)],
+    true,
+    true,
```
Comment (654wak654) [inline addons/cargo/initSettings.sqf:51]: I don't think this setting has to be global, it's more based on individual preference.

Comment (rautamiekka) [reply addons/cargo/initSettings.sqf:51]: There are many personal pref settings already. Just best to allow it be decided on all 3 levels, it's their problem.

Comment (tbeswick96) [reply addons/cargo/initSettings.sqf:51]: Yep, bad copy paste on my end, will amend :)

File: addons/cargo/stringtable.xml [hunk 2]
```diff
@@ -482,5 +482,11 @@
             <Chinesesimp>成功卸货后，重新打开货物菜单。</Chinesesimp>
             <Korean>화물을 성공적으로 내리고 난 다음 화물 메뉴를 열지 결정합니다.</Korean>
         </Key>
+        <Key ID="STR_ACE_Cargo_carryAfterUnload">
+            <English>Carry After Unload</English>
+        </Key>
+        <Key ID="STR_ACE_Cargo_carryAfterUnload_description">
+            <English>Carries or drags the cargo item after unloading</English>
```
Comment (mharis001) [inline addons/cargo/stringtable.xml:489]:
```suggestion
            <English>Controls whether cargo items are carried or dragged after unloading.</English>
```

**Comment by LupusTheCanine** [comment]
Would it be possible to add an ability to prohibit object from being picked up on per class basis?

**Review by oegitmen-avena — APPROVED** [review]

**Review by 654wak654 — APPROVED** [review]

**Comment by markooff** [comment]
**Thx** :) - @tbeswick96  and @PabstMirror for taking care of this issue and, actually,  big problem !

BUT 

there is one thing more . In my opinion (and also opinions of users of my mod PGMLS ) there will be much better - to INTRODUCE a new special SWITCH / Prohibitor - which can be used for a single class - to prevent this class (but ONLY this class/es) for being picked up automatically .

WHY IS IT SUCH IMPORTANT ?

Because your NEW solution (picking solution) still works perfectly - for some certain sorts of classes . Let me just illustrate it: 

https://markooff.net/obrazki/PGM/bledy_army_ACE_itp/blad_cargo_unload_08_2022/Arma3_x64_2022-08-25_16-49-50-24.jpg

https://markooff.net/obrazki/PGM/bledy_army_ACE_itp/blad_cargo_unload_08_2022/Arma3_x64_2022-08-25_16-50-00-75.jpg

https://markooff.net/obrazki/PGM/bledy_army_ACE_itp/blad_cargo_unload_08_2022/Arma3_x64_2022-08-25_16-50-29-42.jpg

...And it works for such object (almost) perfectly !

BUT

for another types of objects (which still could be dragged !) like quads, pontons, canoes, cargo palettes (and so on) - there is one big catastrophy :) with using them , after your last update (picking update) 

Examples : 

BEFORE :

https://markooff.net/obrazki/PGM/bledy_army_ACE_itp/blad_cargo_unload_08_2022/2120189955_preview_Arma3_x64_2022-01-28_03-20-53-38.jpg

https://markooff.net/obrazki/PGM/bledy_army_ACE_itp/blad_cargo_unload_08_2022/2120189955_preview_Arma3_x64_2022-01-28_03-23-28-62.jpg

https://markooff.net/obrazki/PGM/bledy_army_ACE_itp/blad_cargo_unload_08_2022/2120189955_preview_Arma3_x64_2022-01-28_03-23-34-24.jpg

AFTER : 

https://markooff.net/obrazki/PGM/bledy_army_ACE_itp/blad_cargo_unload_08_2022/Arma3_x64_2022-08-25_17-36-08-73.jpg

https://markooff.net/obrazki/PGM/bledy_army_ACE_itp/blad_cargo_unload_08_2022/Arma3_x64_2022-08-25_17-36-12-49.jpg

https://markooff.net/obrazki/PGM/bledy_army_ACE_itp/blad_cargo_unload_08_2022/Arma3_x64_2022-08-25_17-36-41-22.jpg

https://markooff.net/obrazki/PGM/bledy_army_ACE_itp/blad_cargo_unload_08_2022/Arma3_x64_2022-08-25_17-37-36-04.jpg

https://markooff.net/obrazki/PGM/bledy_army_ACE_itp/blad_cargo_unload_08_2022/Arma3_x64_2022-08-25_17-37-48-67.jpg

The same is with other vehicles which act as a "cargo containers" for years , such like :

CH-53 and Quads (unloading) 

https://markooff.net/obrazki/PGM/bledy_army_ACE_itp/blad_cargo_unload_08_2022/ch_53/Arma3_x64_2022-08-12_21-49-55-89.jpg

This situation in spite of the destruction of vehicle(s) will cause another annoying error - to leave out character freezed in DRAG animation ....

https://markooff.net/obrazki/PGM/bledy_army_ACE_itp/blad_cargo_unload_08_2022/ch_53/Arma3_x64_2022-08-12_21-50-02-11.jpg

SUMMARY: 

I will vote AND ASK You for such solution like this described by Lupus the Canine in above post/comment, because in severall mods depended on ACE3 (mine including) we may want - and it's really worth to use this new "picking feature" for SOME classes , but there is no need to use the globally ..... 

THX for Your patience :)

**Comment by tbeswick96** [comment]
@LupusTheCanine @markooff As is the nature of open source, you're welcome to PR the ACE team with your suggestions, but I won't be doing this as part of this PR or at all :)

**Comment by markooff** [comment]
@tbeswick96 - Thanks for Your answer , but I have a few other mods (my own) to carry about, because there are no other person who can do this ... :) 
It's rather part of ACE3 and I think that making such a SWITCH will be much easier for those who work on ACE3 everyday than for me , where I hardly know the code of very few ACE3 modules (which I'm using - within my mods) ..... 
It's a problem of consistency of the whole mod also ... :)

**Pull request merged**